nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/PutBigQueryBatch.java

```diff
@@ -0,0 +1,269 @@
+
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return ImmutableList.<PropertyDescriptor>builder()
+                .addAll(super.getSupportedPropertyDescriptors())
+                .add(DATASET)
+                .add(TABLE_NAME)
+                .add(TABLE_SCHEMA)
+                .add(SOURCE_TYPE)
+                .add(CREATE_DISPOSITION)
+                .add(WRITE_DISPOSITION)
+                .add(MAXBAD_RECORDS)
+                .add(IGNORE_UNKNOWN)
+                .build();
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+                .expressionLanguageSupported(true)
+                .dynamic(true)
+                .build();
+    }
+
+    @Override
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        super.onScheduled(context);
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final Map<String, String> attributes = new HashMap<>();
+
+        final BigQuery bq = getCloudService();
+
+        final String projectId = context.getProperty(PROJECT_ID).evaluateAttributeExpressions().getValue();
+        final String dataset = context.getProperty(DATASET).evaluateAttributeExpressions(flowFile).getValue();
+        final String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+
+        final TableId tableId;
+        if (StringUtils.isEmpty(projectId)) {
+            tableId = TableId.of(dataset, tableName);
+        } else {
+            tableId = TableId.of(projectId, dataset, tableName);
+        }
+
+        final String fileType = context.getProperty(SOURCE_TYPE).getValue();
+
+        String schemaString = context.getProperty(TABLE_SCHEMA).evaluateAttributeExpressions().getValue();
+        Schema schema = BqUtils.schemaFromString(schemaString);
+
+        WriteChannelConfiguration writeChannelConfiguration =
+                WriteChannelConfiguration.newBuilder(tableId)
+                        .setCreateDisposition(JobInfo.CreateDisposition.valueOf(context.getProperty(CREATE_DISPOSITION).getValue()))
+                        .setWriteDisposition(JobInfo.WriteDisposition.valueOf(context.getProperty(WRITE_DISPOSITION).getValue()))
+                        .setIgnoreUnknownValues(context.getProperty(IGNORE_UNKNOWN).asBoolean())
+                        .setMaxBadRecords(context.getProperty(MAXBAD_RECORDS).asInteger())
+                        .setSchema(schema)
+                        .setFormatOptions(FormatOptions.of(fileType))
```

Comment:
In case the input if a CSV file, there are options available to define properties of the dataset (such as header, field delimiter, etc). Can we expose that to the user? or, if not, describe how the data will be processed?